src/python/pants/option/option_types.py

```diff
@@ -113,7 +116,7 @@ def get_flag_options(self, subsystem_cls) -> dict:
         )
 
     @overload
-    def __get__(self, obj: None, objtype: Any) -> OptionsInfo:
+    def __get__(self, obj: None, objtype: Any) -> OptionsInfo | None:
```

Comment:
I'm surprised this type hint doesn't mess things up? Should we instead raise `AssertionError` for accessing an option that isn't registered? Iiuc that's how things work now, that call sites have to check the condition before calling `self.options.foo`

Reply:
Got it. Thanks!